daemon/daemon_test.go

```diff
@@ -0,0 +1,77 @@
+package daemon
+
+import (
+	"testing"
+
+	"github.com/mesg-foundation/core/config"
+	"github.com/mesg-foundation/core/container"
+	"github.com/mesg-foundation/core/container/mocks"
+	"github.com/mesg-foundation/core/x/xnet"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStart(t *testing.T) {
+	cfg, _ := config.Global()
+	c := &mocks.Container{}
+	d := NewContainerDaemon(cfg, c)
+
+	c.On("SharedNetworkID").Return("1", nil)
+	c.On("StartService", d.buildServiceOptions("1")).Return("1", nil)
```

Comment:
having _d.buildServiceOptions("1")_ directly is actually does not check possible mistakes in the creation of _container.ServiceOptions_. it should be manually created here.

Reply:
For buildServiceOptions you have a sperate test. Here it checks if StartService was called